sdk/eventhub/azure-eventhubs/azure/eventhub/aio/error_async.py

```diff
@@ -0,0 +1,86 @@
+from uamqp import errors, compat
+from ..error import EventHubError, EventDataSendError, \
+    EventDataError, ConnectError, ConnectionLostError, AuthenticationError
+
+
+async def _handle_exception(exception, retry_count, max_retries, closable, log):
+    type_name = type(closable).__name__
+    if isinstance(exception, KeyboardInterrupt):
+        log.info("{} stops due to keyboard interrupt".format(type_name))
+        await closable.close()
+        raise
+
+    elif isinstance(exception, (
+            errors.MessageAccepted,
+            errors.MessageAlreadySettled,
+            errors.MessageModified,
+            errors.MessageRejected,
+            errors.MessageReleased,
+            errors.MessageContentTooLarge)
+            ):
+        log.error("Event data error (%r)", exception)
+        error = EventDataError(str(exception), exception)
+        await closable.close(exception)
+        raise error
+    elif isinstance(exception, errors.MessageException):
+        log.error("Event data send error (%r)", exception)
+        error = EventDataSendError(str(exception), exception)
+        await closable.close(exception)
```

Comment:
Does a send error always result in the sender being closed?

Reply:
EventHubConsumer.close( ) no longer closes the connection in when connection is shared. It only closes the link. Anyway I've removed this statement because the link has no problem.